projects/packages/jetpack-mu-wpcom/src/features/private-viewers/use-viewers.ts

```diff
@@ -0,0 +1,131 @@
+import { addQueryArgs } from '@wordpress/url';
+import { useCallback, useEffect, useState } from 'react';
+import wpcomRequest from 'wpcom-proxy-request';
+
+type User = {
+	ID: number;
+	login: string;
+	email: boolean | string;
+	name: string;
+	avatar_URL: string;
+};
+
+type Invite = {
+	invite_key: string;
+	role: string;
+	user: User;
+	is_pending: boolean;
+	invite_date: string;
+	accepted_date: string | null;
+	invited_by: User;
+};
+
+export type Viewer = {
+	inviteId: string;
+	userId: number;
+	status: 'active' | 'pending';
+	name: string;
+	username: string;
+	addedBy: string;
+	inviteDate: string;
+	viewerSince: string;
+	avatarURL: string;
+};
+
+const mapInviteToViewer = ( invite: Invite ): Viewer => {
+	let username;
+	if ( invite.user.login ) {
+		username = invite.user.login;
+	} else if ( invite.user.email ) {
+		username = invite.user.email;
+	} else {
+		username = '';
+	}
+
+	return {
+		inviteId: invite.invite_key,
+		userId: invite.user.ID,
+		status: invite.is_pending ? 'pending' : 'active',
+		name: invite.user.name,
+		username,
+		addedBy:
+			invite.invited_by.name === invite.invited_by.login
+				? invite.invited_by.login
+				: `${ invite.invited_by.name } (${ invite.invited_by.login })`,
+		inviteDate: new Date( invite.invite_date ).toLocaleString( 'default', {
+			dateStyle: 'medium',
+			timeStyle: 'short',
+		} ),
+		viewerSince: invite.accepted_date
+			? new Date( invite.accepted_date ).toLocaleString( 'default', {
+					dateStyle: 'medium',
+					timeStyle: 'short',
+			  } )
+			: '',
+		avatarURL: invite.user.avatar_URL,
+	};
+};
+
+const fetchViewers = async () => {
+	const viewers: Viewer[] = [];
+	const batchSize = 100;
+
+	// Helper function to fetch a single batch of invites.
+	const fetchInvitesBatch = async ( offset: number ) => {
+		const path = addQueryArgs( `/sites/${ window.wpcomPrivateViewers.siteId }/invites`, {
+			offset,
+			number: batchSize,
+			status: 'all',
+			role: window.wpcomPrivateViewers.viewerRole,
+		} );
+		return await wpcomRequest< { invites: Invite[]; found: number } >( {
+			path,
+			apiVersion: '1',
+		} );
+	};
```

Comment:
For some reason this request is not triggered on Atomic sites, even though there are no JS errors 🤔 

<img width="1230" height="961" alt="Image" src="https://github.com/user-attachments/assets/281a6941-2eef-4c13-8dd7-54746e5e63af" />

The `wpcomPrivateViewers` global is defined, so I don't know what's going here.

<img width="1225" height="744" alt="Image" src="https://github.com/user-attachments/assets/11e1abf9-a760-49e6-a63d-29ab26527198" />